todo_app/db.py

```diff
@@ -0,0 +1,139 @@
+		                        {
+		                            "_id": str(ObjectId()),
+                                    "name": "Done",
+                                    "cards": [
+                                                {
+                                                }
+                                            ]
+		                        }
+	                        ]
+                }
+        result = self.get_db().Board.insert_one(board)
+        print("Created Board :" + str(result.inserted_id))
+        return str(result.inserted_id)
+
+    def delete_board(self, board_id):
+        result = self.get_db().Board.delete_one({'_id':board_id})
+        print("DBMongo Deleted Board :" + str(result.deleted_count))
+        return result.deleted_count > 0
+
+
+    def get_cards_on_a_board(self,board_id = os.environ.get('MONGO_BOARD_ID')):
+        cards_on_a_board=[]        
+        for board in self.db.Board.find({'_id' : board_id}):
+            idBoard=board['_id']
+            for status in board['status']:
+                idList=status['_id']
+                for card in status['cards']:
+                    if (len(card) > 0):
+                        id = card['_id']
+                        name = card['name']
+                        due = card['due']
+                        item = {"id": id, "name": name ,"idBoard": idBoard, "idList":idList,"due":due}
+                        cards_on_a_board.append(item)
+                            
+        return cards_on_a_board
+
+    def get_lists_on_a_board(self,board_id = os.environ.get('MONGO_BOARD_ID')):
+        status_on_a_board=[]
+        for board in self.db.Board.find({'_id' : board_id}):
+            idBoard=board['_id']
+            for status in board['status']:
+                id=status['_id']
+                name = status['name']
+                status= {"id": id, "name": name ,"idBoard": idBoard}
+                status_on_a_board.append(status)
+        return status_on_a_board
+
+    def add_a_card_to_ToDoList(self,title, list_id):
+        cardid = str(ObjectId())
+        newCard =   { 
+                        "_id" : cardid,
+                       "name": title,
+                        "due": None
+                    }
+        board_id = os.environ.get('MONGO_BOARD_ID')
+        self.get_db().Board.update_one({'_id':board_id,'status._id':list_id},
+                                      {"$push":{'status.$.cards':newCard}})
+
+    def get_a_card_on_a_board(self,board_id,task_id):
+        cards = self.get_db().Board.aggregate([
+                            { "$match" : { "_id" : board_id} },
+                            { "$unwind" : "$status" },
+                            { "$unwind" : "$status.cards" },
+                            { "$match" : { "status.cards._id" : task_id }},
+                            {"$project": {"_id":1, "status._id":1, "status.cards._id":1, "status.cards.name":1, "status.cards.due":1}}
+                        ])
```

Comment:
Note how complex this is -  a good hint that an alternative data model might have been easier to work with!